internal/command/launch/launch.go

```diff
@@ -535,9 +537,28 @@ func run(ctx context.Context) (err error) {
 		appConfig.SetInternalPort(n)
 	}
 
-	// Finally, write the config
-	if err := appConfig.WriteToDisk(ctx, configFilePath); err != nil {
-		return err
+	// Finally, determine whether we're using Machines and write the config
+	var v2AppConfig *appv2.Config
+	if deployArgs.ForceMachines {
+
+		v2AppConfig, err = appv2.FromDefinition(api.DefinitionPtr(appConfig.Definition))
+		if err != nil {
+			return fmt.Errorf("invalid config: %w", err)
+		}
+		v2AppConfig.AppName = appConfig.AppName
+
+		appConfig.PrimaryRegion = region.Code
```

Comment:
do we need to set this here? I can't tell if it get used later on.

Reply:
If you meant `v2AppConfig.AppName`, that's used when the config is written to disk.
If you meant `appConfig.PrimaryRegion`, I believe it's used in deploy, if you pass `--deploy`.